File: bindings/node/src/zlink/contracts/sockets/dealer_socket.ts
// SPDX-License-Identifier: MPL-2.0

import type { RoutingId } from '../core';
import type { Discovery } from '../service';
import type { PairSocket } from './pair_socket';
import type { RequestOp } from './socket_operations';
import type { DealerSocketOptions } from './socket_options';

export interface DealerSocket extends PairSocket {
  readonly options: DealerSocketOptions;
  setChannelName(channelName: string): void;
  getChannelName(): string;
  setRoutingId(routingId: RoutingId): void;
  getRoutingId(): RoutingId;
  attachDiscovery(discovery: Discovery): void;
  request(): RequestOp;
}

File: bindings/java/src/main/java/systems/zlink/contracts/sockets/MessageSocketContracts/DealerSocket.java
/* SPDX-License-Identifier: MPL-2.0 */

package systems.zlink.contracts.sockets;

import systems.zlink.contracts.core.RoutingId;
import systems.zlink.contracts.messaging.Received;
import systems.zlink.contracts.service.discovery.Discovery;
import systems.zlink.contracts.service.spot.RequestOperation;
import systems.zlink.contracts.service.spot.SendOperation;

public interface DealerSocket extends Socket {
    void bind(String endpoint);
    void connect(String endpoint);
    void unbind(String endpoint);
    void disconnect(String endpoint);
    void disconnectRid(RoutingId routingId);
    void attachDiscovery(Discovery discovery);
    void setChannelName(String channelName);
    String getChannelName();
    void setRoutingId(RoutingId rid);
    RoutingId getRoutingId();
    SendOperation send();
    boolean recv(Received result, RecvFlags flags);
    RequestOperation request();
    @Override DealerSocketOptions options();
}
